lib/spack/spack/test/url_fetch.py

```diff
@@ -244,11 +297,8 @@ def _which(*args, **kwargs):
 
     testpath = str(tmpdir)
     url = 'http://github.com/spack/spack'
-    fetcher = fs.URLFetchStrategy(url=url)
```

Comment:
What's expected to happen in this test now? Is it still needed? Do we need to do a different check?